App/Containers/RegisterCafe.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { Component } from 'react'
+import {
+  SafeAreaView,
+  View,
+  Text,
+  TouchableOpacity,
+  ViewStyle,
+  TextStyle
+} from 'react-native'
+import { Dispatch } from 'redux'
+import { connect } from 'react-redux'
+import { NavigationScreenProps } from 'react-navigation'
+
+import { RootState, RootAction } from '../Redux/Types'
+import { RegisterCafes } from '../features/cafes/reducer'
+
+import { Item, TextileHeaderButtons } from '../Components/HeaderButtons'
+import Icon from '@textile/react-native-icon'
+import CafesList from '../Components/CafesList'
+import CafePeerIdModal from '../Components/CafePeerIdModal'
+import { cafesActions } from '../features/cafes'
+
+import { size, spacing, fontFamily, color } from '../styles'
+
+const Container: ViewStyle = {
+  flex: 1,
+  flexDirection: 'column',
+  justifyContent: 'flex-start',
+  alignItems: 'center'
+}
+
+const ListContainer: ViewStyle = {
+  flex: 1
+}
+
+const Buttons: ViewStyle = {
+  width: '100%',
+  flexDirection: 'row',
+  justifyContent: 'flex-end',
+  alignItems: 'center',
+  padding: spacing._024
+}
+
+interface StateProps {
+  alreadyRegistered: ReadonlyArray<string>
+  registeringCafes: RegisterCafes
+}
+
+interface DispatchProps {
+  register: (url: string, token: string) => void
+}
+
+interface NavProps {
+  openPeerIdModal: () => void
+}
+
+type Props = StateProps & DispatchProps & NavigationScreenProps<NavProps>
+
+interface State {
+  selected?: {
+    url: string
+    token: string
+  }
+  peerIdModalIsVisible: boolean
+}
+
+class RegisterCafe extends Component<Props, State> {
```

Comment:
When I navigate to settings > cafes and tap the button to add a new cafe, I select a cafe to add and tap the arrow button at the bottom of the screen. The cafe is registered, but nothing happens, i just stay on that screen. It's confusing.

Reply:
It fades the button when the registration is still processing, and I just fixed it so it will go back to the cafes screen once registration is finished.